File: infra/src/main/java/com/beat/infra/persistence/exception/PersistenceMappingException.java
package com.beat.infra.persistence.exception;

public final class PersistenceMappingException extends RuntimeException {

	private PersistenceMappingException(String message, RuntimeException cause) {
		super(message, cause);
	}

	public static PersistenceMappingException invalidStoredState(
		String entityName,
		Object entityId,
		RuntimeException cause
	) {
		return new PersistenceMappingException(
			"Invalid stored state for " + entityName + " with id=" + entityId,
			cause
		);
	}
}

File: core/infra/src/main/kotlin/com/beat/infra/persistence/exception/PersistenceMappingException.kt
package com.beat.infra.persistence.exception

internal class PersistenceMappingException private constructor(
    message: String,
    cause: RuntimeException,
) : RuntimeException(message, cause) {
    companion object {
        fun invalidStoredState(
            entityName: String,
            entityId: Any?,
            cause: RuntimeException,
        ): PersistenceMappingException =
            PersistenceMappingException(
                "Invalid stored state for $entityName with id=$entityId",
                cause,
            )
    }
}
